readthedocs/invitations/views.py

```diff
@@ -0,0 +1,112 @@
+"""Invitation views."""
+from django.contrib.auth.mixins import UserPassesTestMixin
+from django.http import Http404, HttpResponseBadRequest, HttpResponseRedirect
+from django.shortcuts import get_object_or_404
+from django.urls import reverse
+from django.views.generic import DeleteView, DetailView
+
+from readthedocs.core.mixins import PrivateViewMixin
+from readthedocs.core.permissions import AdminPermission
+from readthedocs.invitations.models import Invitation
+from readthedocs.organizations.models import Organization, Team, TeamInvite
+
+
+class RevokeInvitation(PrivateViewMixin, UserPassesTestMixin, DeleteView):
```

Comment:
Same question valid for Audit logs here. Does it make sense to mark it as "revoked" instead of deleting it?

Reply:
Agreed, it should be marked "revoked".

Also, we should test that the invitation isn't already used when revoking it. Otherwise, the system can be abused to grant access and cover up the trails.

Reply:
@stsewd proposing to migrate this to a follow-up issue.